### PR TITLE
Feature: REST client patch request

### PR DESCRIPTION
**Pull request by bmaxime-c**
Add http PATCH request on RestClient. ✅
Run dotnet format on the whole solution. ✅
Many typo fixes. ✅

### CODE REVIEW

File: test/AspNetCoreApiSample.IntegrationTests/Resources/TaskResourceTest.cs
Comment (devpro) [inline test/AspNetCoreApiSample.IntegrationTests/Resources/TaskResourceTest.cs:70]: Would it be possible to remove all the added line returns?

File: samples/AspNetCoreApiSample/Utils/PropertiesChangeTracker.cs
Comment (devpro) [inline samples/AspNetCoreApiSample/Utils/PropertiesChangeTracker.cs:10]: Why do we need this file?
I don't see it used in code referencing TaskPatchDto.
(BTW Utils is a too generic word)
It would be best to have the code as much simple as possible.

File: samples/AspNetCoreApiSample/Dto/TaskPatchDto.cs
Comment (devpro) [inline samples/AspNetCoreApiSample/Dto/TaskPatchDto.cs:8]: Remove PropertiesChangedTracker inheritance (not used) and use simple get set properties

**Review by devpro — CHANGES_REQUESTED** [review]
Small but needed changes to be compliant.

File: samples/AspNetCoreApiSample/Utils/PropertiesChangeTracker.cs
Comment (bmaxime-c) [reply samples/AspNetCoreApiSample/Utils/PropertiesChangeTracker.cs:10]: This class allows to track which properties were changed since the instanciation or reinitialization (with AcceptChanges method).
This is useful to differentiate for example null properties because the value was never modified (so in a PATCH behavior, this value will not be modified in repository), from null value which was explicitly set (the value has to be erased in the repository).
There are some libraries which allows this tracking, but they are not free (PostSharp) or need a configuration that seems to be not compliant with hexagonal architecture (eg: TrackerDog).

Comment (devpro) [reply samples/AspNetCoreApiSample/Utils/PropertiesChangeTracker.cs:10]: Interesting!
Would it be possible to add this test scenario in the integration tests? (do a patch with at least one null value then check the values returned by the get)

File: test/AspNetCoreApiSample.IntegrationTests/Resources/TaskResourceTest.cs
Comment (devpro) [inline test/AspNetCoreApiSample.IntegrationTests/Resources/TaskResourceTest.cs:96]: Line ending to remove :)

File: test/AspNetCoreApiSample.IntegrationTests/Resources/TaskResourceTest.cs
Comment (devpro) [inline test/AspNetCoreApiSample.IntegrationTests/Resources/TaskResourceTest.cs:112]: Line ending to remove :)

File: test/AspNetCoreApiSample.IntegrationTests/Resources/TaskResourceTest.cs
Comment (devpro) [inline test/AspNetCoreApiSample.IntegrationTests/Resources/TaskResourceTest.cs:170]: Line ending to remove :)

**Review by devpro — CHANGES_REQUESTED** [review]

**Comment by devpro** [comment]
There is an issue with secret management in the pipeline for pull request coming from forks, I'll bypass the check so the PR can be approve when comments are resolved.

**Comment by bmaxime-c** [comment]
I removed unexpected line endings and added a new test method AspNetCoreApiSampleTaskResourcePatch_WhenModifyOneProperty_ReturnsExpectedObject which verifies that only one property is updated when using patch method.

**Review by devpro — CHANGES_REQUESTED** [review]
@bmaxime-c One missing thing I should better document in the README, can you please update Directory.Build.props file to increase the minor version value?

This is mandatory to have a new version of the NuGet packaqes to be generated.
Once done I'll approve right away the PR and all packages will be generated.

**Comment by bmaxime-c** [comment]
Done, the version number is now 1.3.3

**Pull request merged**
